sdk-setup/src/mb2

```diff
@@ -1799,7 +1799,34 @@ sign_rpms() {
     if [[ -x $sdksign ]]; then
         $sdksign ${rpms}
     else
-        rpmsign --addsign ${rpms}
+        if [[ $OPT_PACKAGE_SIGN_USER ]]; then
+            if ! gpg2 --list-secret-keys "$OPT_PACKAGE_SIGN_USER" &>/dev/null; then
+                notice "Cannot sign packages: No GPG key found for the user '$OPT_PACKAGE_SIGN_USER'."
+                return 1
+            fi
+            local passphrase=
+            if [[ $OPT_PACKAGE_SIGN_PASSPHRASE_FILE ]]; then
+                passphrase=$(<"$OPT_PACKAGE_SIGN_PASSPHRASE_FILE")
+            else
+                passphrase=$OPT_PACKAGE_SIGN_PASSPHRASE
+            fi
+            if ! gpg2 --batch --status-fd 1 --sign --local-user "$OPT_PACKAGE_SIGN_USER" \
+                    --output - --passphrase-fd 3 3<<<"$passphrase" </dev/null &>/dev/null; then
+                if [[ ! $passphrase ]]; then
+                    notice "Cannot sign packages: The selected GPG key is passphrase protected" \
+                        "and no passphrase was specified."
+                else
+                    notice "Cannot sign packages: The given passphrase is not a correct passphrase" \
+                        "for the '$OPT_PACKAGE_SIGN_USER' key."
+                fi
+                return 1
+            fi
+
+            echo -e "${passphrase}\n${passphrase}\n" | rpm --addsign ${rpms} --define "_gpg_name $OPT_PACKAGE_SIGN_USER" \
+                --define "_gpg_sign_cmd_extra_args --batch --passphrase-fd 0"
+        else
+            rpm --addsign ${rpms}
+        fi
     fi
     local rc=$?
     [[ $rc -eq 0 ]] || { notice "Failed to sign packages"; return $rc; }
```

Comment:
> Two times password need because rpmsign ask password twice per file

And we are not limited to packages that produce single RPM, so we'd need the passphrase repeated N*2 times... no way - we need a passphrase-file instead.

Was there any reason why use used `rpm` instead of `rpmsign`? If no, I would prefer using that.

```suggestion
        sign_rpms_native ${rpms}
    fi
    local rc=$?
    [[ $rc -eq 0 ]] || { notice "Failed to sign packages"; return $rc; }
}

sign_rpms_native() {
    local rpms=("$@")

    local tmpdir=
    sign_rpms_native_cleanup() (
        trap 'echo cleaning up...' INT TERM HUP
        if [[ $tmpdir ]]; then
            rm -rf "$tmpdir"
        fi
    )
    trap 'sign_rpms_native_cleanup; trap - RETURN' RETURN
    trap 'return 1' INT TERM HUP

    local extra_args=()

    if [[ $OPT_PACKAGE_SIGN_USER ]]; then
        if ! gpg2 --list-secret-keys "$OPT_PACKAGE_SIGN_USER" &>/dev/null; then
            notice "Cannot sign packages: No GPG key found for the user '$OPT_PACKAGE_SIGN_USER'."
            return 1
        fi

        extra_args+=(--define "_gpg_name $OPT_PACKAGE_SIGN_USER")

        local passphrase=
        if [[ $OPT_PACKAGE_SIGN_PASSPHRASE_FILE ]]; then
            passphrase=$(<"$OPT_PACKAGE_SIGN_PASSPHRASE_FILE")
        else
            passphrase=$OPT_PACKAGE_SIGN_PASSPHRASE
        fi

        if ! gpg2 --batch --status-fd 1 --sign --local-user "$OPT_PACKAGE_SIGN_USER" \
                --output - --passphrase-fd 3 3<<<"$passphrase" </dev/null &>/dev/null; then
            if [[ ! $passphrase ]]; then
                notice "Cannot sign packages: The selected GPG key is passphrase protected" \
                    "and no passphrase was specified."
            else
                notice "Cannot sign packages: The given passphrase is not a correct passphrase" \
                    "for the '$OPT_PACKAGE_SIGN_USER' key."
            fi
            return 1
        fi

        if [[ $passphrase ]]; then
            tmpdir=$(mktemp -d) || return
            chmod 0700 "$tmpdir" || return

            local passphrase_file=$tmpdir/passphrase
            cat >"$passphrase_file" <<<"$passphrase" || return

            extra_args+=(--define "_gpg_sign_cmd_extra_args --batch --passphrase-file '$passphrase_file'")
        fi
    fi

    rpmsign --addsign "${rpms[@]}" "${extra_args[@]}" </dev/null
```